src/BlockInfo.h

```diff
@@ -47,13 +47,14 @@ class cBlockInfo
 
 	// tolua_end
 
-	inline static cBlockHandler * GetHandler      (BLOCKTYPE a_Type) { return Get(a_Type).m_Handler.get();       }
-
 	/** Creates a default BlockInfo structure, initializes all values to their defaults */
 	cBlockInfo();
 
+	/** Gets the blockhandler for the given block type. */
+	static cBlockHandler * GetHandler(BLOCKTYPE a_Type);
```

Comment:
Why move this out of line and out of the blockinfo array?

Reply:
The array doesn't store pointers to handlers anymore, and the implementation that calls cBlockHandler::GetBlockHandler is in the cpp where it knows about BlockHandler.h.

Reply:
>The array doesn't store pointers to handlers anymore

Yes, I can see that. I'm asking why.

Reply:
The handlers are static and cBlockHandler knows where they are and pointers can be retrieved by GetBlockHandler, why have an extra step?

Reply:
A switch statement is just another form of lookup table. There is no extra step.

Reply:
Sure but you have to generate the table at runtime which then takes up some memory. BlockInfo is going to be made entirely stateless so we won't have a cBlockInfoArray anymore. The current fixed std:;array<256> isn't going to play nicely with 1.13+ or warn us if we forget to handle a block type in a switch statement.